src/cloudant/document.py

```diff
@@ -293,36 +348,34 @@ def delete_attachment(self, attachment, headers=None):
         """
         _delete_attachment_
 
-        Delete an attachment from a document
+        Delete an attachment from a document and refresh the local document
+        object
 
         :param str attachment: the attachment file name
         :param dict headers: Extra headers to be sent with request
 
         """
-        attachment_url = posixpath.join(self.document_url, attachment)
-
         # need latest rev
-        doc_resp = self.r_session.get(self.document_url)
-        doc_resp.raise_for_status()
-        doc_json = doc_resp.json()
+        self.fetch()
+        attachment_url = posixpath.join(self.document_url, attachment)
         if headers is None:
-            headers = {'If-Match': doc_json['_rev']}
+            headers = {'If-Match': self['_rev']}
         else:
-            headers['If-Match'] = doc_json['_rev']
+            headers['If-Match'] = self['_rev']
 
         resp = self.r_session.delete(
             attachment_url,
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
is the `self.fetch()` on 359 insufficient? It seems a shame to do two GETs in such quick succession.